client/cmd/resource/upload.go

```diff
@@ -0,0 +1,193 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/goto/salt/log"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"github.com/goto/optimus/client/cmd/internal/connection"
+	"github.com/goto/optimus/client/cmd/internal/logger"
+	"github.com/goto/optimus/client/local"
+	"github.com/goto/optimus/client/local/model"
+	"github.com/goto/optimus/client/local/specio"
+	"github.com/goto/optimus/config"
+	"github.com/goto/optimus/core/resource"
+	pb "github.com/goto/optimus/protos/gotocompany/optimus/core/v1beta1"
+)
+
+const (
+	uploadTimeout    = time.Minute * 30
+	defaultBatchSize = 10
+)
+
+type uploadCommand struct {
+	logger     log.Logger
+	connection connection.Connection
+
+	clientConfig   *config.ClientConfig
+	configFilePath string
+
+	namespaceName string
+	resourceNames []string
+	batchSize     int
+
+	resourceSpecReadWriter local.SpecReadWriter[*model.ResourceSpec]
+}
+
+// NewUploadCommand initializes command for uploading a single resource
+func NewUploadCommand() *cobra.Command {
+	uploadCmd := &uploadCommand{
+		logger: logger.NewClientLogger(),
+	}
+
+	cmd := &cobra.Command{
+		Use:     "upload",
+		Short:   "Upload a resource to server",
+		Long:    heredoc.Doc(`Apply local changes to destination server which includes creating/updating resources`),
+		Example: "optimus resource upload -R [resource1,resource2] -n []",
```

Comment:
We are providing single namespace in the command, are we accepting a list of namespaces here ?

Reply:
no, it should only a single namespace

Reply:
updated